packages/react-compose/src/types.ts

```diff
@@ -141,6 +141,17 @@ export type ComposePreparedOptions<TProps = {}, TInputState = any, TParentState
 // Component types
 //
 
+export interface ComponentProps {
+  // Removing these props:
+  // design - use style or className instead
+
+  as?: React.ElementType;
+
+  className?: string;
+
+  root?: SlotProp<React.HTMLAttributes<HTMLElement>>;
```

Comment:
i don't think `root` is quite right. what if we want to be accurate?
`SlotProp<React.ButtonHTMLAttributes<HTMLButtonElement>>`

Reply:
These `typings` are at the compose layer so I tried to do the most general as possible. I've since changed this to use generic typings.